applications_tests/lethe-fluid-matrix-free/generators/turbulent_taylor_couette_generator.prm

```diff
@@ -0,0 +1,161 @@
+  set grid arguments     = 3.14159265359 : 0.5 : 1.0 : 5 : 4 : true
+  set initial refinement = 1
+end
+
+#---------------------------------------------------
+# FEM
+#---------------------------------------------------
+
+subsection FEM
+  set velocity order = 2
+  set pressure order = 2
+end
+
+#---------------------------------------------------
+# Post-Processing
+#---------------------------------------------------
+
+subsection post-processing
+  set verbosity                    = verbose
+  set calculate kinetic energy     = true
+  set calculate enstrophy          = true
+  set calculate average velocities = false
+  set initial time                 = 0.
+end
+
+#---------------------------------------------------
+# Boundary Conditions
+#---------------------------------------------------
+
+subsection boundary conditions
+  set number                = 3
+  set fix pressure constant = true
+  set time dependent        = true
+  subsection bc 0
+    set type = function
+    set beta = 1
+    subsection u
+      set Function expression = -y
+    end
+    subsection v
+      set Function expression = x
+    end
+    subsection w
+      set Function expression = 0
+    end
+  end
+  subsection bc 1
+    set type = noslip
+  end
+  subsection bc 2
+    set type               = periodic
+    set id                 = 2
+    set periodic_id        = 3
+    set periodic_direction = 2
+  end
+end
+
+#---------------------------------------------------
+# Non-Linear Solver Control
+#---------------------------------------------------
+
+subsection non-linear solver
+  subsection fluid dynamics
+    set tolerance = 1e-3
+    set verbosity = quiet
+  end
+end
+
+#---------------------------------------------------
+# Linear Solver Control
+#---------------------------------------------------
+
+subsection linear solver
+  subsection fluid dynamics
+    set method            = gmres
+    set max iters         = 100
+    set relative residual = 1e-4
+    set minimum residual  = 1e-7
+    set preconditioner    = gcmg
+    set verbosity         = quiet
+
+    # MG parameters
+    set mg verbosity       = quiet
+    set mg min level       = -1
+    set mg level min cells = 16
+
+    # Smoother
+    set mg smoother iterations     = 10
+    set mg smoother eig estimation = true
+
+    # Eigenvalue estimation parameters
+    set eig estimation smoothing range = 5
+    set eig estimation cg n iterations = 20
+    set eig estimation verbosity       = quiet
+
+    # Coarse-grid solver
+    set mg coarse grid solver                 = gmres
+    set mg gmres max iterations               = 2000
+    set mg gmres tolerance                    = 1e-7
+    set mg gmres reduce                       = 1e-4
+    set mg gmres max krylov vectors           = 30
+    set mg gmres preconditioner               = ilu
+    set ilu preconditioner fill               = 1
+    set ilu preconditioner absolute tolerance = 1e-10
+    set ilu preconditioner relative tolerance = 1.00
```

Comment:
Could we replace this by a direct solver? So that it is the same as the other one?
`    # Coarse-grid solver
    set mg coarse grid solver = direct
`

Reply:
Done